lib/provider/window_dimensions_provider.dart

```diff
@@ -1,29 +1,37 @@
 import 'dart:ui';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:localsend_app/provider/persistence_provider.dart';
 import 'package:screen_retriever/screen_retriever.dart';
 import 'package:window_manager/window_manager.dart';
 
 //Records are a better alternative, but they are currently experimental
-typedef WindowDimensions = Map<String, OffsetBase?>?;
+typedef WindowDimensions = Map<String, OffsetBase?>;
 
-class WindowDimensionProvider {
+final windowDimensionProvider = Provider<WindowDimensionsController>((ref) {
```

Comment:
Yep, this is the riverpod-way. Nice :)

Reply:
Incrementally we got there! I was testing things out with the StateController concept with currentDimensions, but ultimately I opted to just write up the Provider proper and be done with it 😂